### PR TITLE
arrpc: 3.2.0 -> 3.3.0

### PR DESCRIPTION
**Pull request by NotAShelf**
## Description of changes

Bumps arrpc from 3.2.0 to 3.3.0, adds changelogs and removes the `with lib;` anti-pattern in the meta section. I was not able to tag the changelogs as upstream does not tag releases.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by NotAShelf** [comment]
thought ofborg pinged maintainers by default...

CC @Anomalocaridid for review

**Review by Anomalocaridid — APPROVED** [review]
Looks good to me!

**Review by fufexan — APPROVED** [review]
LGTM

File: pkgs/by-name/ar/arrpc/package.nix [hunk 1]
```diff
@@ -1,41 +1,31 @@
...
-  npmDepsHash = "sha256-vxx0w6UjwxIK4cgpivtjNbIgkb4wKG4ijSHdP/FeQZ4=";
+  npmDepsHash = "sha256-YlSUGncpY0MyTiCfZcPsfcNx3fR+SCtkOFWbjOPLUzk=";
 
   dontNpmBuild = true;
 
-  meta = with lib; {
+  meta = {
+    # ideally we would do blob/${version}/changelog.md here
+    # upstream does not tag releases
+    changelog = "https://github.com/OpenAsar/arrpc/blob/main/changelog.md";
```
Comment (sikmir) [inline pkgs/by-name/ar/arrpc/package.nix:24]:
```suggestion
    changelog = "https://github.com/OpenAsar/arrpc/blob/${src.rev}/changelog.md";
```

File: pkgs/by-name/ar/arrpc/package.nix [hunk 2]
```diff
@@ -1,41 +1,31 @@
 { lib
 , buildNpmPackage
 , fetchFromGitHub
-, fetchpatch
-}:
-
-buildNpmPackage {
+}: buildNpmPackage {
```
Comment (sikmir) [inline pkgs/by-name/ar/arrpc/package.nix:4]:
```suggestion
}: buildNpmPackage rec {
```

Comment (NotAShelf) [reply pkgs/by-name/ar/arrpc/package.nix:4]: I don't see a reason to use rec.

Comment (sikmir) [reply pkgs/by-name/ar/arrpc/package.nix]: But you can't use src.rev without rec.

Comment (NotAShelf) [reply pkgs/by-name/ar/arrpc/package.nix]: I can inherit it inside src, that's not exactly a problem. But I've went ahead with rec anyway.

**Review by Anomalocaridid — APPROVED** [review]
Still LGTM

**Comment by itslychee** [comment]
Result of `nixpkgs-review pr 286659` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>arrpc</li>
  </ul>
</details>

**Review by itslychee — APPROVED** [review]

**Pull request merged**